sphinx_needs/functions/common.py

```diff
@@ -170,14 +170,15 @@ def copy(
         need = needs[need_id]
 
     if filter:
+        location = (
+            (need["docname"], need["lineno"]) if need and need["docname"] else None
+        )
         result = filter_needs_view(
             needs,
```

Comment:
this is problematic, and in fact any use of `needs` in dynamic functions, because we do not know if we really need to account on whether we are calling it from the "resolution phase", i.e. dynamic functions in need fields, or the analysis phase, e.g. dynamic functions in need content